### ISSUE TITLE
JABBA Structural Upgrade

### ISSUE THREAD

**Issue by JasonMcRay** [JasonMcRay/InfiTech-1.7#283]
Cobalt -> Obsidian -> Diamond

**Comment by DarknessShadow** [comment]
Well if we are changing the sturctural upgrades we could also make it better for gregtech
something like this
wood(or rubber?) -> (wrought) iron -> gold -> cobalt -> steel -> aluminium -> (obsidian ->) emerald -> diamond -> titanium -> tungstensteel -> iridium -> osmium -> neutronium

this is nice a nice config

    # Maximum tier to generate crafting recipes for
    I:maxCraftableTier=7

**Comment by Xavion3** [comment]
That list is rather ridiculously long, tone things down a tad would be my suggestion but the basic idea works. Something like wood -> bronze -> wrought iron -> cobalt -> steel -> obsidian -> diamond -> aluminium -> titanium -> tungstensteel -> iridium

That's a total of 11 upgrades, enough to put it to 16x the upgrades of current barrels making them easily the best high density storage system if you've teched up enough to get the higher tiers. If it happens I'd say add an alternate recipe for storage upgrades to decrease the stupid huge cost they'd have, maybe the current recipe with palladium plates added makes 9x storage or something? It'd give a use for palladium which is currently totally useless as far as I can tell and in terms of difficulty to acquire is only from platinum/iridium veins I'm pretty sure, you could use other materials though.

**Comment by DarknessShadow** [comment]
Those would be 22 structural upgrades each doubles the provided slots
(2^22) * 64 = 268.435.456 stacks = 17.179.869.184 items = 17 billion items
compared to the gregtech quantum chest
mk1 = 1,6 million items
mk2 = 9,7 million items
mk3 = 58,3 million items
mk4 = 349 million items
mk5 = 2 billion items

Hmmm maybe its a bit long ... but we can fix it.
I think the last 5 structural upgrades should just require 4x mk1 / mk2 / mk3 / mk4 / mk5 quantum chest to craft.

**Comment by Xavion3** [comment]
Also the game might break if you tried to do so many, if they're only using ints to store the count it will break for instance as an int can only count up to about 2 billion. That's why I suggested only having 11, you really don't need every tier of metals. That makes it max at about 8 million items (assuming stacks of 64), of course that's failing to account for the storage upgrades. The 17 billion items would require over 8 million iron ingots and redstone for instance, tens of millions of wood and cobble as well. The 11 upgrades would only have about 2k upgrade slots, not the 4 million the 22 upgrades would grant.

So yeah, 10 or 11 upgrades should be your max amount most likely, why I suggested the ones I did.
